### PR TITLE
[Event Hubs] Fix backwards compatibility with new vs old SDK checkpointing

### PR DESCRIPTION
**Pull request by m-redding**
The API updates in https://github.com/Azure/azure-sdk-for-net/pull/41829 created an unintentional side effect where processors running different versions of the SDK can no longer work together. Since offset was removed from the checkpoint position, new checkpoints no longer write the offset to the checkpoint. To preserve the formatting, a default value of long.MinValue was passed in. This leaves processors using the old SDK version to continue to try positioning at offset long.MinValue and failing.

This change sets the offset in the blob to string.Empty, so that when the processor pulls in the checkpoint, it will be able to fallover to the sequence number. The logging infrastructure now writes "" to the logs when an offset or sequence number is not provided.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/BlobCheckpointStoreInternal.Diagnostics.cs [hunk 1]
```diff
@@ -95,19 +95,19 @@ static BlobCheckpointStoreInternal()
         /// <param name="eventHubName">The name of the specific Event Hub the checkpoint is associated with, relative to the Event Hubs namespace that contains it.</param>
         /// <param name="consumerGroup">The name of the consumer group the checkpoint is associated with.</param>
         /// <param name="clientIdentifier">The unique identifier of the client that authored this checkpoint.</param>
-        /// <param name="sequenceNumber">The sequence number associated with the checkpoint.</param>
-        /// <param name="replicationSegment">The replication segment associated with the checkpoint.</param>
-        /// <param name="offset">The offset associated with the checkpoint.</param>
+        /// <param name="sequenceNumber">The sequence number associated with the checkpoint. This value is set to <c>-1</c> if one is not provided.</param>
```
Comment (m-redding) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/BlobCheckpointStoreInternal.Diagnostics.cs:98]: These could also be defaulted to long.MinValue. My rationale for changing to -1 is that when looking at the logs the long.MinValue looks somewhat random and seems like it's coming from somewhere, not that it's just nothing was provided.

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/BlobEventStoreEventSource.cs [hunk 2]
```diff
@@ -321,13 +321,13 @@ protected BlobEventStoreEventSource() : base(EventSourceName)
                                                   string consumerGroup,
                                                   string clientIdentifier,
                                                   string errorMessage,
-                                                  long sequenceNumber,
+                                                  string sequenceNumber,
                                                   int replicationSegment,
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/BlobEventStoreEventSource.cs:325]: Should we just make `replicationSegment` a string as well?   We know that it's basically a dummy value when there's no GeoDR enabled.   I don't feel strongly about it, but thought it worth mentioning.

**Review by jsquire — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/BlobEventStoreEventSource.cs (hunk 2 above)
Comment (m-redding) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/BlobEventStoreEventSource.cs]: updated - I had them all as string initially but I don't remember why I changed it

**Pull request merged**